slides/04_modules.md

```diff
@@ -37,25 +37,37 @@ Unported License, [http://creativecommons.org/licenses/by-sa/4.0/](http://creati
 # How to use modules (1/2)
 
 - The syntax is simple: `module command module-name`
-    - For example: `module load gromacs-env`
-    - [List of most common commands](https://docs.csc.fi/computing/modules/#module-commands-table)
+   - For example: `module load gromacs-env`
+   - [List of most common commands](https://docs.csc.fi/computing/modules/#module-commands-table)
 - You can't just load all the modules because of the dependencies
 - If you try to load a module that is not available, you will get an error message saying so 
 
 # How to use modules (2/2)
- - These commands will help you figuring out the module situation:
-    - `module list`: See the list of modules loaded at the moment
-    - `module avail`: Modules available at the moment (due to depencies -hides modules that can't be loaded atm)
-     - `module spider name`: Search for an application in the list of all existing modules
-     - `module spider name/version`: Gives information on how to load the module (prerequisites etc).
 
-# Conda environments
+- These commands will help you figuring out the module situation:
+   - `module list`: See the list of modules loaded at the moment
+   - `module avail`: Modules available at the moment (due to depencies -hides modules that can't be loaded atm)
+   - `module spider name`: Search for an application in the list of all existing modules
+   - `module spider name/version`: Gives information on how to load the module (prerequisites etc).
+
+# Various kinds of modules
+
+- Some modules contain software that has a graphical user interface
+   - For example chemistry visualisation software [Molden](https://docs.csc.fi/apps/molden/)
+   - Use Puhti Web Interface or [other connection with GUI](https://docs.csc.fi/computing/connecting/#using-graphical-applications)
+- Some modules contain software that provide a command line interface
+   - For example [Allas object storage](https://docs.csc.fi/data/Allas/)
+- Some modules provie you with a collection of eg. Python or R packages
+   - For example [Bioconda](https://docs.csc.fi/apps/bioconda/), [Geoconda](https://docs.csc.fi/apps/geoconda/#using-geoconda) or [Python Data](https://docs.csc.fi/apps/python-data/)
+
+# Conda environments and Python packages
```

Comment:
This is a bit tricky question. I think the original motivation to include conda here is that there are e.g. 40+ biosoftware packages installed by CSC that are not loaded as individual modules, but rather accessed through the bioconda module (load bioconda module, activate the environment). So the point was that there are "hidden" software packages not visible with e.g. "module spider". But it is indeed correct that we don't advocate conda for our customers any more when it can be avoided, and we will migrate our conda-based installations to containers too, at some point.

Reply:
I see. 
And also find this tricky, since the name of our 'geoconda' module would suggest that it is working in the same way, but it's not (nowadays it's a container with conda fully removed after installing packages). 

What do you think about the suggestion I made below about rebranding the topic from conda to python and listing the modules with a hint to conda in bio onda module and adding package

Reply:
I think in order to keep things brief and on-topic, we could pretty much omit any mention of Conda in this slide set. We could add some generic comment in the spirit of "Some software packages are not loaded as their own modules, but are accessed through some other module. See documentation for each software". 

Any mention about making your own installations (conda or not) should go to the installation slides. They are currently bit of a mess, but I'm revising them.

We should also think in what kind of detail we go to the Python modules. We could again just mention that different Python modules are available and documentation for each is available in Docs.

Also: This course is not meant to be bio-oriented, but rather as generic as possible. It may currently seem a bit bio-heavy, because it is largely based on older material (that was for bio audience) and much of it was written by me, Laxman and Maria (and we are all bio people). It would be a good thing if we can get rid of that "historical baggage" as we keep developing the material.

Reply:
Ah yes, I like your suggested sentence. This could go on the module page?

In geoinformatics we also have some modules that are old baggage that can be very misleading when finding modules only via module spider etc commands, eg there is a module called grass but if you want to use newest installation of grass you need to load qgis module, which is all written in docs ( we are in process of cleaning up).

Agree that some of this should go into installation slides.

Let me know if you need some geoinformatics examples to get away from 'only bio' 🙂

Reply:
Thank you for comments! These gave me good guidelines.